File: ch31/App_Code/VB/Listing31_1.vb
Imports Microsoft.VisualBasic
Imports System.Net

Public Class Listing31_1
    Public Function RSSFeedLength(uris As IList(Of Uri)) As Integer
        Dim total As Integer = 0
        For Each uri In uris
            Dim data = New WebClient().DownloadData(uri)
            total += data.Length
        Next
        Return total
    End Function


End Class

File: ch31/App_Code/CS/Listing31_1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Web;

/// <summary>
/// Summary description for Listing01
/// </summary>
public class Listing31_1
{
    public Listing31_1()
    {

    }
    public int RSSFeedLength(IList<Uri> uris)
    {
        int total = 0;
        foreach (var uri in uris)
        {
            var data = new WebClient().DownloadData(uri);
            total += data.Length;
        }
        return total;
    }


}